crates/wdk-build/rust-driver-makefile.toml

```diff
@@ -41,31 +41,79 @@ if is_file ${filepath}
-out = exec --fail-on-error rust-script --base-path ${taskjson.env.CARGO_MAKE_CURRENT_TASK_INITIAL_MAKEFILE_DIRECTORY} ${CARGO_MAKE_CRATE_CUSTOM_TRIPLE_TARGET_DIRECTORY}/cargo-make-script/${task.name}/main.rs %{cli_args}
-assert_eq ${out.code} 0 "[tasks.${task.name}]'s script failed with exit code: ${out.code}\nstdout:\n${out.stdout}\nstderr:\n${out.stderr}\nThe temporary rust-script file is located at ${CARGO_MAKE_CRATE_CUSTOM_TRIPLE_TARGET_DIRECTORY}/cargo-make-script/${task.name}/main.rs"
 
 # Set cargo-make env vars based on output of rust-script
 script_output = trim ${out.stdout}
 if not is_empty ${script_output}
     script_output_array = split ${script_output} \n
-    stdout_first_line = array_get ${script_output_array} 0
-    assert_eq ${stdout_first_line} "FORWARDING ARGS TO CARGO-MAKE:" "[tasks.${task.name}]'s script output did not begin with \"FORWARDING ARGS TO CARGO-MAKE:\". Was `--help` passed as one of the arguments?\nstdout:\n${out.stdout}\nstderr:\n${out.stderr}\nThe temporary rust-script file is located at ${CARGO_MAKE_CRATE_CUSTOM_TRIPLE_TARGET_DIRECTORY}/cargo-make-script/${task.name}/main.rs"
-    array_remove ${script_output_array} 0
+
+    # Search the stdout output of the script, with the following behaviours for each line:
+    # 1. If the line is between the "FORWARDING ARGS TO CARGO-MAKE:" start delimited and the "END OF FORWARDING ARGS TO CARGO-MAKE" end delimiter, update the cargo-make process' environment variables based on the key-value pairs in the line.
+    # 2. If the line is not between the start and end delimiters, print the line normally.
+    looking_for_start_delimiter = set true
     for line in ${script_output_array}
-        parts = split ${line} =
-        key = array_get ${parts} 0
-        value = array_get ${parts} 1
-        set_env ${key} ${value}
+        if ${looking_for_start_delimiter}
+            if eq ${line} "FORWARDING ARGS TO CARGO-MAKE:"
+                looking_for_start_delimiter = set false
+            else
+                # Any output not surrounded by the start and end delimiter lines should be printed normally
+                println ${line}
+            end
+        else
+            if eq ${line} "END OF FORWARDING ARGS TO CARGO-MAKE"
+                looking_for_start_delimiter = set true
+            else
+                # Set cargo-make env_var based on line output
+                parts = split ${line} =
+                key = array_get ${parts} 0
+                value = array_get ${parts} 1
+                set_env ${key} ${value}
+            end
+        end
     end
+    assert ${looking_for_start_delimiter} "A matching \"END OF FORWARDING ARGS TO CARGO-MAKE\" for a \"FORWARDING ARGS TO CARGO-MAKE:\" was not found in script output."
 end
 '''
 
+# This plugin replaces the embedded `CARGO_MAKE_CURRENT_TASK_INITIAL_MAKEFILE_DIRECTORY` in the rust-script doc comment with its evaluated value. See https://github.com/sagiegurari/cargo-make/issues/1081 for more context.
```

Comment:
Nope. I just didn't want to make the change in this PR since scope is already extremely large here. Will address in a followup. Tracking this in https://github.com/microsoft/windows-drivers-rs/issues/171 already